### PR TITLE
Update Dependencies

### PR DESCRIPTION
**Pull request by baffles**
Merging in dependency updates performed by Abe

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -251,11 +251,11 @@ pub struct ApiClient {
...
     pub fn new(config: Box<ClientConfig>) -> ApiClient {
-        let mut client_builder = reqwest::Client::builder();
+        let client_builder = reqwest::Client::builder();
         // the --insecure CLI flag enables this, to disable the CN name check
-        if config.allows_insecure() {
-            client_builder.danger_disable_hostname_verification();
-        }
+        // if config.allows_insecure() {
+        //     client_builder.danger_accept_invalid_certs(true);
+        // }
```
Comment (baffles) [inline src/client.rs]: This looks wrong to me, no clue if this was intentional on Abe's part or not...

Comment (dylemma) [reply src/client.rs]: Yeah you should probably undo this... I assume the `danger_accept_invalid_certs` method still exists, so idk why he'd do this aside from forgetting about it during cleanup.

**Review by dylemma — CHANGES_REQUESTED** [review]
Everything seems fine with the exception of what you already called out.

File: Cargo.toml [hunk 2]
```diff
@@ -11,7 +11,7 @@ url = "1.5.1"
 serde = "1.0"
 serde_json = "1.0"
 serde_derive = "1.0"
-reqwest = "0.8"
+reqwest = { version = "0.9.15", default-features = false }
```
Comment (baffles) [inline Cargo.toml]: @dylemma any idea why 0.9.15 was picked? It's far from the latest version, and it's not even the latest 0.9.x release...

File: src/client.rs (hunk 1 above)
Comment (baffles) [reply src/client.rs]: They've shuffled these functions around a little bit and the docs give some (rightfully so) strongly worded warnings about using them. What exactly is this insecure flag aiming to accomplish? Why do we have/need a flag to disable the hostname check?

It seems like accepting self-signed certificates would be the main thing we need in this area, not a disabling of hostname/certificate checks, especially without a strongly worded warning of our own.

File: Cargo.toml [hunk 3]
```diff
@@ -1,17 +1,17 @@
 [package]
 name = "codedx-client"
 version = "0.2.0"
```
Comment (baffles) [inline Cargo.toml]: I guess version should probably be bumped now?

**Review by dylemma — DISMISSED** [review]

File: Cargo.toml (hunk 3 above)
Comment (dylemma) [reply Cargo.toml]: Ah yeah, might as well bump to 0.3.0

File: src/client.rs [hunk 4]
```diff
@@ -214,30 +212,31 @@ pub type ApiResult<T> = Result<T, ApiError>;
...
+            if status.is_success() {
                 Ok(response)
             } else {
-                ApiErrorMessage::from_body(&mut response).and_then(|response_msg| {
-                    Err(ApiError::NonSuccess(response.status(), response_msg))
+                ApiErrorMessage::from_body(response).and_then(|response_msg| {
+                    Err(ApiError::NonSuccess(status, response_msg))
                 })
             }
         }))
```
Comment (baffles) [inline src/client.rs:235]: I'm not 100% confident that I changed this correctly

Comment (dylemma) [reply src/client.rs:235]: Seems perfectly fine to me

**Review by dylemma — APPROVED** [review]

**Pull request merged**